src/app/modules/item/components/item-children-list/item-children-list.component.html

```diff
@@ -0,0 +1,45 @@
+<ul
+  *ngIf="children.length > 0; else noChildrenSection"
+  class="data-list-container sub-item"
+  [ngClass]="{'activity-type': type === 'activity'}"
+>
+  <li *ngFor="let item of children; let i = index">
+    <div class="activity-progress" *ngIf="type === 'activity'">
+      <alg-score-ring
+        [diameter]="30"
+        [currentScore]="item.result?.score ?? 0"
+        [isValidated]="item.result?.validated ?? false"
+        [bestScore]="item.bestScore"
+        *ngIf="item.result"
+      ></alg-score-ring>
+    </div>
+    <div
+      class="item-content"
+      (click)="click(item)"
+      [ngClass]="{ locked: item.isLocked }"
+    >
+      <a class="item-title alg-link">
+        {{ i + 1 }}. {{ item.string.title }}
+      </a>
+      <span *ngIf="item.isLocked" class="item-status">
+        <i *ngIf="item.isLocked" class="fa fa-lock"></i>
+      </span>
+      <alg-skill-progress
+        *ngIf="type === 'skill' && item.result"
+        style="width: 25%;"
+        type="thick-horizontal"
+        [bestScore]="item.bestScore"
+        [currentScore]="item.result.score"
+      ></alg-skill-progress>
+    </div>
+  </li>
+</ul>
+
+<ng-template #noChildrenSection>
```

Comment:
This block, as it is, will not be compatible with a reuse in chapter children. Possibly, the message could be a parameter set by the parent component.

Reply:
Fixed